patterns/singleton/Singleton-lazy.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+
+public sealed class Student
+{
+    // Properties of the student
+    public string Name { get; set; }
+    public int Age { get; set; }
+    public string Grade { get; set; }
+
+    // Private static instance variable using Lazy initialization
+    private static readonly Lazy<Student> instance = new Lazy<Student>(() => new Student("Default",0,"None"));
+    
+    // Public static property to access the singleton instance
+    public static Student Instance
+    {
+        get
+        {
+            return instance.Value;
+        }
+    }
+
+    // Constructor
+    private Student(string name, int age, string grade)  // Making private constructor to prevent instantiation from outside
+    {
+        Name = name;
+        Age = age;
+        Grade = grade;
+    }
+
+    // Method to update student details
+    public void UpdateDetails(string Name, int Age, string Grade)
```

Comment:
Is it typical in C# to use Pascal Case (i.e. uppercase on all words, including first) rather than camel Case (i.e. lowercase on first word, uppercase on all others) for method names?

Reply:
@ash-teach Yes, it is the part of c# naming convention.
We need to use Pascal case for the Methods and I will change the parameter names to camelCase.
Again thanks for your feedback and pointing out the errors.
![Standard c#](https://github.com/nic-dgl104-winter-2024/pattern-library/assets/54470199/05eac1d9-62c5-4fa4-ad1c-929f540dfbc2)

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names